packages/headless/src/controllers/core/quickview/headless-core-quickview.ts

```diff
@@ -110,37 +148,64 @@ export function buildCoreQuickview(
   const getState = () => engine.state;
   const controller = buildController(engine);
   const {result, maximumPreviewSize} = props.options;
-  const uniqueId = result.uniqueId;
+
+  const getUniqueIdFromPosition = () => {
+    const {resultsWithPreview, position} = getState().resultPreview;
+    return resultsWithPreview[position];
+  };
+
+  dispatch(preparePreviewPagination({results: getState().search.results}));
+
+  const onFetchContent = (uniqueId: string) => {
+    props.options.onlyContentURL
+      ? dispatch(
+          updateContentURL({
+            uniqueId,
+            requestedOutputSize: maximumPreviewSize,
+            buildResultPreviewRequest,
+            path,
+          })
+        )
+      : dispatch(
+          fetchResultContent({
+            uniqueId,
+            requestedOutputSize: maximumPreviewSize,
+          })
+        );
+    if (fetchResultContentCallback) {
+      fetchResultContentCallback();
+    }
+  };
 
   return {
     ...controller,
 
     fetchResultContent() {
-      props.options.onlyContentURL
-        ? dispatch(
-            updateContentURL({
-              uniqueId,
-              requestedOutputSize: maximumPreviewSize,
-              buildResultPreviewRequest,
-              path,
-            })
-          )
-        : dispatch(
-            fetchResultContent({
-              uniqueId,
-              requestedOutputSize: maximumPreviewSize,
-            })
-          );
-      if (fetchResultContentCallback) {
-        fetchResultContentCallback();
-      }
+      onFetchContent(result.uniqueId);
+    },
+
+    next() {
+      dispatch(nextPreview());
+      onFetchContent(getUniqueIdFromPosition());
```

Comment:
Yeah, good thinking Benjamin !

I did test manually, and there is indeed a potential problem, especially true when you need to load a very large quickview. There is no mistmatch between state, but what's actually happening is that the `currentResult` will update out of order, which does create a very weird result for the end user in the quickview pagination section.

The solution is two fold I think: 

In this PR, we can very simply make it so that in the reducer we return early in `nextPreview` and `previousPreview` when `isLoading=true`.

Secondly, we should add a loading gif/animation when a quickview is taking "a long time", as can be the case for very very big PDF.

As for it blocking pressing next button very very quickly, I think with a combination of loading gif that appears quickly, it gives good feedback to the end user that the UI is working, and should be more than acceptable from a UX standapoint.